src/integrations/cleanup-integration.php

```diff
@@ -520,4 +530,34 @@ protected function cleanup_orphaned_from_table( $table, $column, $limit ) {
 		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared, WordPress.DB.PreparedSQL.NotPrepared, WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Reason: Already prepared.
 		return $wpdb->query( "DELETE FROM $table WHERE {$column} IN( " . \implode( ',', $orphans ) . ' )' );
 	}
+
+	/**
+	 * Deletes rows from the indexable table where the source is no longer there.
+	 *
+	 * @param string $source_table The source table which we need to check the indexables against.
+	 * @param string $source_identifier The identifier which the indexables are matched to.
+	 * @param string $object_type The indexable object type.
+	 * @param int    $limit The limit we'll apply to the delete query.
+	 *
+	 * @return int|bool The number of rows that was deleted or false if the query failed.
+	 */
+	protected function clean_indexables_for_object_type_and_source_table( $source_table, $source_identifier, $object_type, $limit ) {
```

Comment:
I don't agree since this does not remove every orphaned indexable but just specific combinations of source and object type that is why I chose this bigger name :)

Reply:
Fair enough :)